daemon/qrexec-daemon.c

```diff
@@ -256,53 +261,60 @@ void init(int xid)
             // invalid or negative number
             startup_timeout = MAX_STARTUP_TIME_DEFAULT;
     }
-    signal(SIGUSR1, sigusr1_handler);
-    signal(SIGCHLD, sigchld_parent_handler);
-    switch (pid=fork()) {
-    case -1:
-        perror("fork");
-        exit(1);
-    case 0:
-        break;
-    default:
-        if (getenv("QREXEC_STARTUP_NOWAIT"))
-            exit(0);
-        if (!opt_quiet)
-            fprintf(stderr, "Waiting for VM's qrexec agent.");
-        for (i=0;i<startup_timeout;i++) {
-            sleep(1);
-            if (!opt_quiet)
-                fprintf(stderr, ".");
-            if (i==startup_timeout-1) {
+
+    if (!opt_direct) {
+        signal(SIGUSR1, sigusr1_handler);
+        signal(SIGCHLD, sigchld_parent_handler);
+        switch (pid=fork()) {
+            case -1:
+                perror("fork");
+                exit(1);
+            case 0:
                 break;
-            }
+            default:
+                if (getenv("QREXEC_STARTUP_NOWAIT"))
+                    exit(0);
+                if (!opt_quiet)
+                    fprintf(stderr, "Waiting for VM's qrexec agent.");
+                for (i=0;i<startup_timeout;i++) {
+                    sleep(1);
+                    if (!opt_quiet)
+                        fprintf(stderr, ".");
+                    if (i==startup_timeout-1) {
+                        break;
+                    }
+                }
+                fprintf(stderr, "Cannot connect to '%s' qrexec agent for %d seconds, giving up\n", remote_domain_name, startup_timeout);
+                exit(3);
         }
-        fprintf(stderr, "Cannot connect to '%s' qrexec agent for %d seconds, giving up\n", remote_domain_name, startup_timeout);
-        exit(3);
     }
+
     close(0);
-    snprintf(qrexec_error_log_name, sizeof(qrexec_error_log_name),
-         "/var/log/qubes/qrexec.%s.log", remote_domain_name);
-    umask(0007);        // make the log readable by the "qubes" group
-    logfd =
-        open(qrexec_error_log_name, O_WRONLY | O_CREAT | O_TRUNC,
-         0660);
-
-    if (logfd < 0) {
-        perror("open");
-        exit(1);
-    }
 
-    dup2(logfd, 1);
-    dup2(logfd, 2);
+    if (getuid() == 0) {
```

Comment:
Currently qrexec-daemon is _not_started as root, so this isn't a good condition here (it is good only in qrexec-agent). Maybe put it under `-D` option too (daemon -> log to file, no daemon -> log to stderr)?

Reply:
Yeah, I think it's better than another option. Done.

Reply:
You've pushed to #19 branch (which is fine and also needed), but not here. I'm ok with that, but github won't notice when it's merged.